src/m365/base/SpoCommand.ts

```diff
@@ -98,18 +98,18 @@ export default abstract class SpoCommand extends Command {
     return true;
   }
 
-  public action(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
-    auth
-      .restoreAuth()
-      .then((): void => {
-        if (auth.service.connected && AuthType[auth.service.authType] === AuthType[AuthType.Secret]) {
-          cb(new CommandError(`SharePoint does not support authentication using client ID and secret. Please use a different login type to use SharePoint commands.`));
-          return;
-        }
+  public async action(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      await auth.restoreAuth();
+    }
+    catch (error: any) {
+      throw new CommandError(error);
+    }
+
+    if (auth.service.connected && AuthType[auth.service.authType] === AuthType[AuthType.Secret]) {
```

Comment:
This should go in the `try` block. If restoring auth fails, there's no point in executing this check, because auth will be undefined/invalid.

Reply:
If restoring the auth fails, an error is thrown in the catch block and thus the method is interrupted. The highlighted piece of code won't run in that case.

Also this piece of code throws an error, if I put this in the `try` block, this error will be wrapped twice as `CommandError`. Although I think it can be done this way:

```typescript
try {
  await auth.restoreAuth();

  if (auth.service.connected && AuthType[auth.service.authType] === AuthType[AuthType.Secret]) {
    throw `SharePoint does not support authentication using client ID and secret. Please use a different login type to use SharePoint commands.`;
  }
}
catch (error: any) {
  throw new CommandError(error);
}

super.action(logger, args);
```

Your call what's the most ideal solution.

Reply:
Good point. One one hand, I like your current solution as it keeps the code inside `try` to the minimum and avoid unnecessary nesting. On the other hand, if we ever change what's inside `catch` the rest of the code will run so we need to pay extra attention to it. Let's stick with your current solution and ensure we have proper tests in place to validate the desired behavior.